src/Parallafka/MessagesByKey.cs

```diff
@@ -14,17 +16,34 @@ namespace Parallafka
     /// </summary>
     internal class MessagesByKey<TKey, TValue>
     {
-        private readonly Dictionary<TKey, Queue<IKafkaMessage<TKey, TValue>>> _messagesToHandleForKey;
+        private readonly CancellationToken _cancellationToken;
+        private readonly Dictionary<TKey, Queue<KafkaMessageWrapped<TKey, TValue>>> _messagesToHandleForKey;
         private readonly TaskCompletionSource _completedSource;
         private bool _completed;
 
-        public MessagesByKey()
+        /// <summary>
+        /// Creates a new instance of <see cref="MessagesByKey{TKey,TValue}"/>
+        /// </summary>
+        /// <param name="cancellationToken">When the token is cancelled, no more messages will be processed</param>
+        public MessagesByKey(CancellationToken cancellationToken)
         {
+            this._cancellationToken = cancellationToken;
             this._messagesToHandleForKey = new();
             this._completedSource = new();
             this.Completion = this._completedSource.Task;
         }
 
+        public object GetStats()
+        {
+            lock (this._messagesToHandleForKey)
+            {
+                return new
+                {
+                    MessageCountsByKey = this._messagesToHandleForKey.ToDictionary(kvp => kvp.Key, kvp => kvp.Value?.Count ?? 0)
```

Comment:
Can this be done without the lock since it's just a stats route and there's no need for strict consistency? Granted, lock contention from GetStats calls would probably be rare